ziti-agent/cmd/webhook/webhook.go

```diff
@@ -137,8 +117,131 @@ func serve(w http.ResponseWriter, r *http.Request, admit admitHandler) {
 	}
 }
 
-func serveZitiTunnelSC(w http.ResponseWriter, r *http.Request) {
-	serve(w, r, newDelegateToV1AdmitHandler(handleZitiTunnelAdmission))
+func zitiClientImpl() (*rest_management_api_client.ZitiEdgeManagement, error) {
+
+	// parse ziti admin certs to synchronously (blocking) create a ziti identity
+	zitiAdminIdentity, err := tls.X509KeyPair(zitiAdminCert, zitiAdminKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(zitiAdminIdentity.Certificate) == 0 {
+		err := fmt.Errorf("no certificates found in TLS key pair")
+		return nil, err
+	}
+
+	parsedCert, err := x509.ParseCertificate(zitiAdminIdentity.Certificate[0])
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(4).Infof("Parsed client certificate - Subject: %v, Issuer: %v", parsedCert.Subject, parsedCert.Issuer)
+	klog.V(4).Infof("Loading CA bundle, size: %d bytes", len(zitiCtrlCaBundle))
+	klog.V(5).Infof("CA bundle content: %s", string(zitiCtrlCaBundle))
+
+	certPool := x509.NewCertPool()
+	if !certPool.AppendCertsFromPEM(zitiCtrlCaBundle) {
+		err := fmt.Errorf("failed to append CA certificates from PEM")
+		return nil, err
+	}
+
+	cfg := zitiedge.Config{ApiEndpoint: zitiCtrlMgmtApi,
+		Cert:       parsedCert,
+		PrivateKey: zitiAdminIdentity.PrivateKey,
+		CAS:        *certPool,
+	}
+
+	cfg.CABundle = zitiCtrlCaBundle
+	zc, err := zitiedge.Client(&cfg)
+
+	return zc, err
+
+}
+
+func serveZitiTunnel(w http.ResponseWriter, r *http.Request) {
+
+	kc, err := k.Client()
+	if err != nil {
+		w.Header().Set("Content-Type", "application/json")
+		err = fmt.Errorf("failed to initilize cluster client: %v", err)
+		klog.Error(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	zc, err := zitiClientImpl()
+	if err != nil {
+		w.Header().Set("Content-Type", "application/json")
+		err = fmt.Errorf("failed to initilize ziti client: %v", err)
+		klog.Error(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	zh := newZitiHandler(
+		&clusterClient{client: kc},
+		&zitiClient{client: zc},
+		&zitiConfig{
+			ZitiType:        zitiTypeTunnel,
+			VolumeMountName: "sidecar-ziti-identity",
+			LabelKey:        "tunnel.openziti.io/enabled",
+			RoleKey:         zitiRoleKey,
+			Image:           sidecarImage,
+			ImageVersion:    sidecarImageVersion,
+			ImagePullPolicy: sidecarImagePullPolicy,
+			IdentityDir:     sidecarIdentityDir,
+			Prefix:          sidecarPrefix,
+			LabelDelValue:   "false",
+			LabelCrValue:    "true",
+			ResolverIp:      clusterDnsServiceIP,
+			RouterConfig:    routerConfig{},
+		},
+	)
+	serve(w, r, newAdmitHandler(zh.handleAdmissionRequest))
```

Comment:
Something's not right because when I tried to test a CREATE event the admission response is sent before the ziti identity is created.